third-party/realdds/include/realdds/dds-stream-base.h

```diff
@@ -38,12 +39,14 @@ class dds_stream_base : public std::enable_shared_from_this< dds_stream_base >
     // Init functions can only be called once!
     void init_profiles( dds_stream_profiles const & profiles, int default_profile_index = 0 );
     void init_options( dds_options const & options );
+    void enable_metadata() { _metadata_enabled = true; }
```

Comment:
Two things: if you can enable you should be able to disable.
But I don't think you should be able to change after construction. We should enforce that this is possible only before the stream profiles have been initialized or similar.

Reply:
I agree that it should be set during initialization and cannot be changed later but I did not want to add a third parameter to a constructor that so many types inherit.
I added a check that it will be called before `init_profiles`